clients/amqp-connection/src/main/java/org/eclipse/hono/client/amqp/tracing/AmqpMessageExtractAdapter.java

```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.client.amqp.tracing;
+
+import java.util.AbstractMap;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Map.Entry;
+import java.util.Objects;
+
+import org.apache.qpid.proton.amqp.messaging.ApplicationProperties;
+import org.apache.qpid.proton.message.Message;
+
+import io.opentracing.propagation.TextMap;
+
+/**
+ * An adapter for extracting properties from an AMQP 1.0 message.
```

Comment:
is it worth mentioning that we extract the properties from the AMQP message's application-properties?